File: 2150-find-all-lonely-numbers-in-the-array/2150-find-all-lonely-numbers-in-the-array.cpp
class Solution {
public:
    vector<int> findLonely(vector<int>& nums) {
        unordered_map<int,int> mp;
        vector<int>ans;
        for(int i = 0;i<nums.size();i++)
        {   int val = nums[i];
            if(mp[val] == 0)
            {
                mp[val] = 1;
            }
            else
            {
                mp[val]++;
            }
        }
        for(int i = 0;i<nums.size();i++)
        {
            int val = nums[i];
            if(mp[val] == 1 && mp[val+1] == 0 && mp[val-1] == 0)
            {
                ans.push_back(val);
            }
        }
    return ans;
    }
};

File: 2150-find-all-lonely-numbers-in-the-array/2150-find-all-lonely-numbers-in-the-array.java
class Solution {
    public List<Integer> findLonely(int[] nums) {
        int[] arr = new int[1000001] ;
        List<Integer> result = new ArrayList<>() ;
        for(int i : nums){
            if(i>0)
            arr[i-1]++ ;
            arr[i]++ ;
            if(i<1000000)
            arr[i+1]++ ;
        }
        for(int i: nums){
            if(arr[i]==1){
                result.add(i) ;
            }
        }
        return result ;
    }
}